src/modules/form-dyn.js

```diff
@@ -0,0 +1,303 @@
+								ctn: elem,
+								form: form,
+								data: data
+							});
+						}
+						App.mediator.notify('recaptcha.forceUpdate', {
+							ctn: elem,
+							form: form
+						});
+					}
+				},
+				fieldsOptions: {
+					rules: {
+						date: {
+							datetime: {
+								dateOnly: true,
+								message: formElem.attr('data-msg-date-invalid'),
+								notValid: formElem.attr('data-msg-date-invalid')
+							}
+						},
+						document: {
+							format: {
+								message: formElem.attr('data-msg-doc-invalid')
+							}
+						},
+						image: {
+							format: {
+								message: formElem.attr('data-msg-img-invalid')
+							}
+						}
+					},
+					emptinessEvents: null,
+					previewEvents: null,
+					onFocus: function (data) {
+						App.mediator.notify('formDyn.inputFocus', _.assign({
+							ctn: elem,
+							form: form
+						}, data));
+					},
+					onBlur: function (data) {
+						App.mediator.notify('formDyn.inputBlur', _.assign({
+							ctn: elem,
+							form: form
+						}, data));
+					},
+					onKeyup: function (data) {
+						App.mediator.notify('formDyn.keyup', _.assign({
+							ctn: elem,
+							form: form
+						}, data));
+					},
+					onChangeOrInput: function (data) {
+						App.mediator.notify('formDyn.inputChange', _.assign({
+							ctn: elem,
+							form: form
+						}, data));
+					},
+					onInited: function (data) {
+						//Attach Date picker if class is present
+						if (data.field.find('.js-date-picker').length) {
+							var it = data.field.find('.js-date-picker');
+							var dateFormat = $('html').attr('lang') == 'fr' ?
+								'dd-mm-yy' :
+								'mm-dd-yy';
+							it.datepicker();
+							it.datepicker('option', 'dateFormat', dateFormat);
+						}
+					}
+				},
+				onError: function (data) {
+					App.mediator.notify('formDyn.validationError', {
+						results: data.results,
+						ctn: elem,
+						form: form
+					});
+				},
+				onBeforePost: function (e) {
+					var cancel = false;
+					App.mediator.notify('recaptcha.execute', {
+						ctn: elem,
+						form: form
+					}, function (key, ret) {
+						cancel = ret;
+					});
+					return cancel;
+				}
+			});
+			elem.data('component', form);
+			App.mediator.notify('formDyn.init', {
+				form: form,
+				ctn: elem
+			});
+			elem.data('component', form);
+			
+			btnSubmit = elem.find(sels.formSubmit);
+			btnSubmit.on($.click, onFormSubmitClick);
+			forms.push(form);
+		});
+	};
+
+	var onPageEnter = function (key, data) {
+		page = $(data.page.key());
+		initForms(page);
+	};
+
+	var onArticleEnter = function (key, data) {
+		var article = $(data.article);
+		initForms(article);
+	};
+	
+	var onPageLeave = function (key, data) {
+		page = $(data.page.key());
+		resetForms(page);
+		resetForms(preFooter);
+	};
+
+	var onArticleLeave = function (key, data) {
+		var article = $(data.article);
+		resetForms(article);
+		resetForms(preFooter);
+	};
+	
+	var init = function () {
+		initForms(preFooter);
```

Comment:
Why do we have a preFooter in this module? It seems too much site-specific to include it in the default code IMO.

Reply:
We want to keep the scope of form dyn inside the actual page, not #site but we have, in a lot of projects, a newsletter form that uses form-dyn inside the footer.

Reply:
Yes, I get it but _preFooter_? I've only seen this nomenclature in 1 project recently. I think it could be added on a per project basis, but if we want to always init it, we should at least use _footer_ instead of _prefooter_. (Still think its a gamble to say there will be a newsletter there)

Reply:
I can agree with that. We should simply change prefooter to footer then.

Reply:
+1